### DISCUSSION TITLE
[BP#9] Limiter le nombre de requêtes HTTP

### DISCUSSION THREAD

**Discussion by devcrafting** [cnumr/best-practices#203] [Bonnes pratiques] [answered]
Elle me semble bien. Je remplacerais juste SGBD/R par "bases de données" (qui ne sont pas que relationnelles désormais...)

**Comment by devcrafting** [comment]
Cf. https://github.com/cnumr/best-practices/discussions/220 (cas particulier sous forme d'une BP à part)

**Comment by ACTLEM** [comment]
En phase avec le changement de wording pour les bases de données.

**Comment by ldesgrange** [comment]
Les deux premières phrases de cette BP sont un peu redondantes, on pourrait en écrire qu'une seule.

Ensuite elle parle de 2 choses :
- le nombre de requêtes HTTP
- la taille des fichiers

La taille des fichiers n'étant pas le sujet de cette BP (et couvert par d'autres), je pense qu'il faudrait supprimer ces mentions ainsi que l'exemple glyphe vs images.

Et pour le nombre de requêtes HTTP, cette BP est principalement basé sur le fait que les requêtes soient faites séquentiellement, problème largement résolu par HTTP/2 (et qui rend les utilisations de sprite et concaténation de fichiers nettement moins pertinente (en tout cas de la façon dont c'était fait traditionnellement)).

Je ne sais pas si cette BP est finalement très pertinente en l'état et la remplacer par une BP "Utiliser HTTP/2" ?

**Reply by ACTLEM** [reply]
Il y a déjà une BP sur HTTP/2 qui est créée pour la V4 (https://github.com/cnumr/best-practices/pull/189).
Comme je l'ai déjà écrit dans la discussion https://github.com/cnumr/best-practices/discussions/220, je ne suis pas à l'aise avec cette BP qui est plus du résultat attendu et est très lié à d'autres BP qui permettent d'atteindre de résultat (et c'est d'autant plus vrai avec la BP sur HTTP/2). Bref, j'ai dû mal à voir son intérêt. Donc plutôt en phase avec ta remarque.

**Reply by devcrafting** [reply]
donc ce serait plutôt une suppression :), vu qu'on a déjà le reste

**Reply by Fred-artwai** [reply]
Bonjour, je pense la même chose cette BP être datée HTTP/1. Si il doit y a voir une limite du nombre de requêtes puisque le titre de la BP est "Limiter le nombre de requêtes HTTP" il faut la définir : 50 ? 100 ? Mais sur quels critères ? Si on ne sait définir cette limite effectivement il faut supprimer cette BP.

**Reply by tbroyer** [reply]
Sur EcoMeter, on avait fixé le seuil à 25% de la moyenne de HTTPArchive (106 requêtes par page à l'époque, donc un seuil à 26 requêtes: https://gitlab.com/ecoconceptionweb/ecometer/-/blob/901e669804337775e4b6c64d1f12c3c93cea5c0c/webpage-analyzer/rules/180-limit-http-requests.js#L6-11)
GreenIT-Analysis utilise 2 seuils: à 26 et 40 requêtes: https://github.com/cnumr/GreenIT-Analysis/blob/5dd5d094267782a51573d00f196c067cbf386b1a/script/rules/HttpRequests.js#L18-L19

**Reply by ACTLEM** [reply]
@tbroyer, @rlemaire quel est votre avis sur cette BP (la garder malgré les autres BP qui amènent à ce résultat ou la supprimer)?

**Reply by rlemaire** [reply]
C'est vrai que c'est plus un titre regroupant plusieurs pratique : on veut limiter les requêtes dont on fait A, B, C, ...

**Reply by tbroyer** [reply]
Ça ne me choque pas qu'on ait des BP assez générales au début du livre qui sont déclinées par la suite en BP plus ciblées.

En gros, pour reprendre les critères EcoIndex, on veut limiter le nombre de requêtes HTTP, limiter la "complexité" de la page côté client (taille du DOM dans EcoIndex) et limiter le poids transféré sur le réseau. Et pour faire tout ça, on va optimiser les images, minifier les JS et CSS, les compresser, etc.
Si on n'a que les BP "ciblées", on a le risque de passer à côté de certains trucs (BP manquante) et surtout de noyer le message général ; si on n'a que la BP générale, elle devient énorme si on essaie d'y lister tout ce qu'on peut mettre en œuvre.
…donc avoir les 2 c'est bien.

**Reply by DocRoms** [reply]
Perso, je garderais bien cette BP, Le HTTP/2 améliore les choses... par domaine. Lorsque l'on a un nombre trop élevé de ressources sur des domaines différents, on se tape quand même toute la phase de résolution DNS + SSL + connexion pour toutes les ressources des domaines n'ayant téléchargés encore aucune ressources.

On l'observe très bien sur les waterfall des sites ayant une tonnes de ressources. 

Par contre, je suis aussi pour le changement de wording, et l'arrêt d'exemple et d'explications relatifs à d'autres BP :)

**Reply by ldesgrange** [reply]
Limiter le nombre de domaines c'est le sujet de la [BP 18](https://github.com/cnumr/best-practices/blob/main/chapters/BP_018_fr.md).

**Reply by ACTLEM** [reply]
Ok, pour résumer on garderait cette BP assez générale mais on l'améliorerait pour éviter qu'elle soit une redite d'autres BP (enlever les références aux autres BP) + correction de wording?

**Reply by ACTLEM** [reply]
Si on supprime la BP20 (https://github.com/cnumr/best-practices/discussions/223), on pourrait garder la référence au sprite css (encore utile dans certains cas: images décoratives, HTTP1 et besoin de forte optimisation)

**Reply by ACTLEM** [reply]
Des objections sur mes deux dernières remarques?

**Reply by florinesueur** [reply]
Aucune objection pour moi.

**Reply by tbroyer** [reply]
SGTM

**Reply by rlemaire** [reply]
Idem

**Comment by ACTLEM** [comment, marked as answer]
Suite aux retours positifs, j'ai créé l'issue pour modification: https://github.com/cnumr/best-practices/issues/298
